src/App.tsx

```diff
@@ -45,6 +47,12 @@ function App() {
         }
     }, [todoList, isLoading]);
 
+    useEffect(() => {
+        if (!isLoading) {
+            return createBalloons(balloonContainerRef.current);
+        }
+    }, [isLoading]);
+
     return isLoading ? (
         <h2>Loading...</h2>
```

Comment:
interesting logic. This invites a funny race condition, which you do address in the util by checking if the ref is `null`

While this works, likely a cleaner approach would be to create a new BalloonContainer component and pass children to it (the h2, Form, and TodoList), then you'd have access to some lifecycle hooks